lib/client/api.go

```diff
@@ -1540,6 +1540,8 @@ func (tc *TeleportClient) NewWatcher(ctx context.Context, watch types.Watch) (ty
 
 // WithRootClusterClient provides a functional interface for making calls
 // against the root cluster's auth server.
+// Deprecated: Prefer reusing auth clients instead of creating at worst two
+// clients for a single function.
```

Comment:
I'm not convinced deprecating this function is the right call. What should the callers do if these are literally one-off?

Looking at the callers of `WithRootClusterClient`, in many cases there is nothing to share the client with. 

The notable exceptions are:
- `sendAccessRequestAndWaitForApproval` (2 usages)
- `executeAccessRequest` and `getAccessRequest` (`executeAccessRequest` has 2 usages and calls `getAccessRequest`)

Instead of deprecating this function, can we instead:
- optimise the use cases above directly (separate PR),
- expand godoc for `WithRootClusterClient` noting the acceptable use cases (no feasible way to share the client) and the correct function to use otherwise.

Reply:
> It only takes 11 lines of code to get a root auth client so the benefits this function provides are minimal.

Inlining the deprecated function at call sites is not going to improve the performance, but it _will_ increase code duplication.

Reply:
The amount of dialing stays the same with inlined function though?